web_ui/ui.go

```diff
@@ -84,10 +84,19 @@ func configureWebResource(engine *gin.Engine) error {
 	engine.GET("/view/*path", func(ctx *gin.Context) {
 		path := ctx.Param("path")
 
+		// If the path is a directory indicate that we are looking for the index.html file
 		if strings.HasSuffix(path, "/") {
 			path += "index.html"
 		}
 
+		// If path doesn't have extension, is not a directory, and has a index file, redirect to index file
+		if !strings.Contains(path, ".") && !strings.HasSuffix(path, "/") {
+			if _, err := webAssets.ReadFile("frontend/out" + path + "/index.html"); err == nil {
+				ctx.Redirect(http.StatusFound, "/view/"+path+"/")
```

Comment:
I'm actually a bit confused here. I though we should not redirect user to `index.html` if their path has `/` suffix, like `/view/login/` and we should append `index.html` to the user path internally to find the correct file whenever there's only a `/` ?

For requests without a trailing slash, and is not a file name, like `/view/login`, we want to permanently redirect them to `/view/login/`.

For requests with a valid file name (that `webAssets.ReadFile` gives a positive answer), we can serve the file directly.

This way, the path is cleaner and more readable for users but we are still able to serve `*index.html` file which won't hurt SEO.

Reply:
Plus, for the `path` parameter, we might want to clean it first to prevent directory traversal attack:

```go
safePath := path.Clean("/" + pathParam)
```